mm2src/adex_cli/src/adex_config.rs

```diff
@@ -0,0 +1,160 @@
+}
+
+impl Display for AdexConfigImpl {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        if !self.is_set() {
+            return writeln!(f, "adex configuration is not set");
+        }
+        writeln!(
+            f,
+            "mm2 RPC URL: {}",
+            self.rpc_uri.as_ref().expect("Expected rpc_uri is set")
+        )?;
+        writeln!(f, "mm2 RPC password: *************")?;
+        Ok(())
+    }
+}
+
+impl AdexConfigImpl {
+    #[cfg(test)]
+    pub(super) fn new(rpc_password: &str, rpc_uri: &str) -> Self {
+        Self {
+            rpc_password: Some(rpc_password.to_string()),
+            rpc_uri: Some(rpc_uri.to_string()),
+        }
+    }
+
+    #[cfg(not(test))]
+    pub(super) fn read_config() -> Result<AdexConfigImpl> {
+        let config = AdexConfigImpl::from_config_path()?;
+        if config.rpc_password.is_none() {
+            warn!("Configuration is not complete, no rpc_password in there");
+        }
+        if config.rpc_uri.is_none() {
+            warn!("Configuration is not complete, no rpc_uri in there");
+        }
+        Ok(config)
+    }
+
+    fn is_set(&self) -> bool { self.rpc_uri.is_some() && self.rpc_password.is_some() }
+
+    pub(super) fn get_config_dir() -> Result<PathBuf> {
+        let project_dirs = ProjectDirs::from(PROJECT_QUALIFIER, PROJECT_COMPANY, PROJECT_APP)
+            .ok_or_else(|| error_anyhow!("Failed to get project_dirs"))?;
+        let config_path: PathBuf = project_dirs.config_dir().into();
+        fs::create_dir_all(&config_path)
+            .map_err(|error| error_anyhow!("Failed to create config_dir: {config_path:?}, error: {error}"))?;
+        Ok(config_path)
+    }
+
+    fn get_config_path() -> Result<PathBuf> {
+        let mut config_path = Self::get_config_dir()?;
+        config_path.push(ADEX_CFG);
+        Ok(config_path)
+    }
+
+    fn from_config_path() -> Result<AdexConfigImpl> {
+        let config_path = Self::get_config_path()?;
+
+        if !config_path.exists() {
+            warn_bail!("Config is not set")
+        }
+        Self::read_from(&config_path)
+    }
+
+    fn write_to_config_path(&self) -> Result<()> {
+        let config_path = Self::get_config_path()?;
+        self.write_to(&config_path)
+    }
+
+    fn read_from(cfg_path: &Path) -> Result<AdexConfigImpl> {
+        let adex_path_str = cfg_path.to_str().unwrap_or("Undefined");
+        let adex_cfg_file = fs::File::open(cfg_path)
+            .map_err(|error| error_anyhow!("Failed to open: {adex_path_str}, error: {error}"))?;
+
+        serde_json::from_reader(adex_cfg_file)
+            .map_err(|error| error_anyhow!("Failed to read adex_cfg to read from: {adex_path_str}, error: {error}"))
+    }
+
+    fn write_to(&self, cfg_path: &Path) -> Result<()> {
+        let adex_path_str = cfg_path
+            .to_str()
+            .ok_or_else(|| error_anyhow!("Failed to get cfg_path as str"))?;
+        rewrite_json_file(self, adex_path_str)
```

Comment:
The file permissions of this config file should be set to 660 to disallow reading by other users.

Reply:
Thank you! [done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1913/files)